File: scripts/clean.py
#!/usr/bin/env python3
# /// script
# requires-python = ">=3.13"
# ///
"""Clean the project: remove caches, build artifacts, and venv."""

import shutil
from pathlib import Path

ROOT = Path(__file__).parent.parent

DIRS_TO_REMOVE = [
    ".venv",
    ".mypy_cache",
    ".ruff_cache",
    ".pytest_cache",
]

GLOB_PATTERNS = [
    "**/__pycache__",
    "**/*.egg-info",
    "**/.temp",
    "**/dist",
    "**/build",
]


def remove(path: Path) -> None:
    if path.exists():
        shutil.rmtree(path)
        print(f"  removed  {path.relative_to(ROOT)}")


def main() -> None:
    print("Cleaning project...\n")

    print("Removing top-level directories...")
    for name in DIRS_TO_REMOVE:
        remove(ROOT / name)

    print("\nRemoving cached/build artifacts...")
    for pattern in GLOB_PATTERNS:
        for path in sorted(ROOT.glob(pattern), reverse=True):
            remove(path)

    print("\nDone.")


if __name__ == "__main__":
    main()

File: demo/scripts/clean.ts
// scripts/clean.ts
// Clean all dev-generated and cache files in the ElectroBun demo app.
//
// Run with:
//   bun run clean
//   bun scripts/clean.ts
//
// This removes build artifacts, Vite output, generated Python runtime,
// caches, etc. These are all gitignored (see .gitignore).

import { $ } from "bun"
import { rm } from "fs/promises"
import { existsSync } from "fs"
import { resolve, join } from "path"

const ROOT = resolve(import.meta.dir, "..")

const DIRS_TO_REMOVE = [
  "build",
  "dist",
  "build/lib",
  "node_modules",
  ".cache",
  "artifacts"
]

async function removeDir(name: string) {
  const fullPath = join(ROOT, name)
  if (existsSync(fullPath)) {
    await rm(fullPath, { recursive: true, force: true })
    console.log(`  removed  ${name}`)
  }
}

async function main() {
  console.log("Cleaning demo dev artifacts...\n")

  console.log("Removing top-level generated directories...")
  for (const dir of DIRS_TO_REMOVE) {
    await removeDir(dir)
  }

  console.log("\nRemoving cache artifacts...")
  // ElectroBun cache (inside node_modules)
  const electrobunCache = join(ROOT, "node_modules/electrobun/.cache")
  if (existsSync(electrobunCache)) {
    await rm(electrobunCache, { recursive: true, force: true })
    console.log("  removed  node_modules/electrobun/.cache")
  }

  // Any leftover *.tsbuildinfo (TypeScript incremental builds)
  try {
    await $`find ${ROOT} -name "*.tsbuildinfo" -type f -delete 2>/dev/null`.quiet()
    console.log("  removed  *.tsbuildinfo files")
  } catch {
    // ignore if none or find not perfect
  }

  // Clean any other common temp/caches that might be left
  const extraGlobs = [
    "**/.temp",
    "**/tmp"
  ]
  for (const pattern of extraGlobs) {
    try {
      await $`find ${ROOT} -path "*/${pattern}" -type d -exec rm -rf {} + 2>/dev/null`.quiet()
    } catch {
    }
  }

  console.log("\nDone. Run `bun install` if you also want to refresh node_modules.")
}

main().catch((err) => {
  console.error("Clean failed:", err)
  process.exit(1)
})
